plugin/trino-clickhouse/src/main/java/io/trino/plugin/clickhouse/ClickHouseClient.java

```diff
@@ -241,6 +243,11 @@ public ClickHouseClient(
         JdbcTypeHandle bigintTypeHandle = new JdbcTypeHandle(Types.BIGINT, Optional.of("bigint"), Optional.empty(), Optional.empty(), Optional.empty(), Optional.empty());
         this.connectorExpressionRewriter = JdbcConnectorExpressionRewriterBuilder.newBuilder()
                 .addStandardRules(this::quoted)
+                .map("$equal(left: varchar, right: varchar)").to("left = right")
```

Comment:
Can we have an additional checks for the data types which are mapped to varchar i.e `STRING` in case of Clickhouse is mapped to `VARCHAR` -

Reply:
Looks like it's not something we have for other connectors too. 
We can document it as a limitation.

Reply:
we do, see Postgres and enum for example.

Reply:
`ClickHouseClient#toColumnMapping`, as many others,  has 2 switches to choose mapping implementation, first on `typeHandle.jdbcTypeName()` - native ClickHouse type, and the second on `typeHandle.jdbcType()` - the generic jdbc type.
Mapping from ClickHouse `String` or `FixedString` to Trino `Varchar` based on `jdbcTypeName`, so it means native type name.
And it means that any other ClickHouse native types are not mistakenly mapped to Trino `Varchar`, because they cannot 'leak' via generic `jdbcType`. 

If we are talking about `getForcedMappingToVarchar` or `getUnsupportedTypeHandling` the logic is following
```
        if (getUnsupportedTypeHandling(session) == CONVERT_TO_VARCHAR) {
            return mapToUnboundedVarchar(typeHandle);
        }
```
where `mapToUnboundedVarchar` has `DISABLE_PUSHDOWN`.

Basing on the above considerations we are safe from pushing down unexpected types.

Reply:
I think `DISABLE_PUSHDOWN` is used at a different places i.e it used along with `TupleDomain` while  `ConnectorExpressionRewrite` has a different code path. Ideally we might need to have a unified source of truth which should inform us if the column is natively mapped or forcefully mapped so we could pushdown the filter accordingly for both that framework. Currently we miss that piece of information so we might need to extract it out. 

Other solution we could use is instead of using this generic re-write rule i.e `"$equal(left: varchar, right: varchar)"` we could have an explicit rule for RewriteEquals, RewriteLike etc which would work only on varchar which has this `typeHandle.jdbcTypeName()` mapped - so in this case we could be sure on which data types are supported and then we could focus on improving the framework

Reply:
https://github.com/trinodb/trino/pull/23558